### ISSUE TITLE
Increasing Robustness by Transformations

### ISSUE THREAD

**Issue by AlanPeng0897** [LukasStruppek/Plug-and-Play-Attacks#5]
Hi Lukas, thanks for this interesting paper, I have two questions about this part.
1. As mentioned in the paper, to adapt the generated images to P_tar(x), you considered some standard image transformations. How could you figure out what kind of transformations are desirable? 
2. You include random transformation to reduce the risk that the attack generates mis-
leading images.  But is the optimization still feasible since random transformations like random cropping are non-differentiable.

**Comment by LukasStruppek** [comment]
Hi Alan,

1.)  We simply took standard random transformations with relatively strong randomness in them. But the specific choice of the transformations did not have a huge impact. So besides random cropping, other transformations, e.g., random rotation, can have similar effects on the results. Since we wanted to avoid biasing the results by selecting transformations that maximize the success, we simply stick with the simple random cropping and fixes cropping parameters.
2.) Yes, the optimization is still possible since most transformations are indeed differentiable (in PyTorch). The reason is that most of the transformations are realized by using simple matrix manipulation techniques that allow gradient computation.

Best,
Lukas

**Comment by AlanPeng0897** [comment]
Thank you for the explanation! Regarding the first point, I am still confused. In your code, you center crop the generated 1024x1024 image into 800x800 to remove some background, and then resize it to the target image size of 224x224 to align with P_tar. If we don't have so much information about the target domain (except for the resolution), how can we determine which image transformation to use to adapt to P_tar?

Best,
Alan

**Comment by LukasStruppek** [comment]
Yes, you can estimate the cropping resolution by selecting different cropping values and then measure the convergence speed (target confidence)during optimization for different classes. If the cropping approximately meets the target distribution, the convergence speed in terms of confidence values for the targeted classes increases much faster. So the parameters can be estimated by a simple grid search and do not require the ground-truth data. We also found the cropping hyperparameter not to be very sensitive, so roughly estimating the value should be sufficient and even without the cropping, the results are overall still reasonable.

Best,
Lukas

**Comment by AlanPeng0897** [comment]
Thanks!
